plugin.py

```diff
@@ -18,21 +20,48 @@ class LspPyrightPlugin(NpmClientHandler):
     server_directory = "language-server"
     server_binary_path = os.path.join(server_directory, "node_modules", "pyright", "langserver.index.js")
 
-    @classmethod
-    def additional_variables(cls) -> Optional[Dict[str, str]]:
-        variables = super().additional_variables() or {}
-        variables.update(
-            {
-                "sublime_py_files_dir": os.path.dirname(sublime.__file__),
-            }
-        )
-
-        return variables
-
     @classmethod
     def install_in_cache(cls) -> bool:
         return False
 
     @classmethod
     def minimum_node_version(cls) -> Tuple[int, int, int]:
         return (12, 0, 0)
+
+    @classmethod
+    def on_client_configuration_ready(cls, configuration: Dict[str, Any]) -> None:
+        super().on_client_configuration_ready(configuration)
+
+        if get_setting("st_plugin_development_mode"):
+            configuration["settings"]["python.analysis.extraPaths"].extend(cls.find_package_dependency_dirs())
+            configuration["settings"]["python.analysis.extraPaths"].extend(
+                [
+                    "$packages",
+                    os.path.dirname(sublime.__file__),
+                ]
+            )
+
+    @staticmethod
+    def find_package_dependency_dirs() -> List[str]:
+        dep_dirs = []  # type: List[str]
+        dep_versions = ["all", "st4", "st3", "st2"]
+        packages_path = sublime.packages_path()
+
+        for path in os.listdir(packages_path):
+            test_package = os.path.join(packages_path, path)
+
+            # is the package a dependency?
+            if not (
+                os.path.isfile(os.path.join(test_package, ".sublime-dependency"))
+                or os.path.isfile(os.path.join(test_package, "dependency-metadata.json"))
+            ):
+                continue
+
+            for version in dep_versions:
+                test_version = os.path.join(test_package, version)
+
+                if os.path.isdir(test_version):
+                    dep_dirs.append(test_version)
+                    break
```

Comment:
All this information is already present in `sys.path`. Check it in the (3.3) console.

Reply:
```
Python 3.3>>> import sys
Python 3.3>>> sys.path
['/opt/sublime_text/Lib/python3.3.zip', '/opt/sublime_text/Lib/python33', '/opt/sublime_text/Lib/python3', '/home/raoul/.config/sublime-text-3/Lib/python33', '/home/raoul/.config/sublime-text-3/Packages', '/home/raoul/.config/sublime-text-3/Packages/pygments/all', '/home/raoul/.config/sublime-text-3/Packages/sublime_lib/st3', '/home/raoul/.config/sublime-text-3/Packages/lsp_utils/st3', '/home/raoul/.config/sublime-text-3/Packages/wcwidth/all', '/home/raoul/.config/sublime-text-3/Packages/coverage/st3_linux_x64', '/home/raoul/.config/sublime-text-3/Packages/ptyprocess/all', '/home/raoul/.config/sublime-text-3/Packages/backrefs/st3', '/home/raoul/.config/sublime-text-3/Packages/bracex/st3', '/home/raoul/.config/sublime-text-3/Packages/markupsafe/all', '/home/raoul/.config/sublime-text-3/Packages/pathlib/all', '/home/raoul/.config/sublime-text-3/Packages/pymdownx/st3', '/home/raoul/.config/sublime-text-3/Packages/pyte/all', '/home/raoul/.config/sublime-text-3/Packages/python-markdown/st3', '/home/raoul/.config/sublime-text-3/Packages/pyyaml/st3', '/home/raoul/.config/sublime-text-3/Packages/python-jinja2/all', '/home/raoul/.config/sublime-text-3/Packages/mdpopups/st3', '/home/raoul/.config/sublime-text-3/Packages/wcmatch/st3']
```

Reply:
Noticed a small problem about `sys.path`. For example, on my setup, `sys.path` outputs the following ordered list.

```py
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Installed Packages\\Restructured Text (RST) Snippets.sublime-package\\wcwidth', 
'C:\\Program Files\\Sublime Text\\Lib\\python3.3.zip', 
'C:\\Program Files\\Sublime Text\\Lib\\python33', 
'C:\\Program Files\\Sublime Text\\Lib\\python3', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Lib\\python33', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Packages', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Packages\\pygments\\all', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Packages\\python-toml\\st3', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Packages\\sublime_lib\\st3', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Packages\\lsp_utils\\st3', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Packages\\backrefs\\st3', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Packages\\bracex\\st3', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Packages\\coloraide\\st3', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Packages\\MARKUP~1\\all', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Packages\\natsort\\all', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Packages\\pathlib\\all', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Packages\\pymdownx\\st3', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Packages\\PYTHON~1\\st3', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Packages\\pyyaml\\st3', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Packages\\requests\\all', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Packages\\typing\\st3', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Packages\\PYTHON~2\\all', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Packages\\wcmatch\\st3', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Packages\\jsonschema\\all', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Packages\\mdpopups\\st3', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Installed Packages\\Codec.sublime-package\\pybase62', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Installed Packages\\Codec.sublime-package\\idna', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Installed Packages\\EditorConfig.sublime-package'
```

Noticed this
```
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Packages', 
'C:\\Users\\Clover\\AppData\\Roaming\\Sublime Text\\Packages\\pygments\\all', 
...
```

All package deps actually won't work because `Packages/` is at the first. So if you do `import pygments`, Pyright will try to find it in `Packages/pygments/` and found nothing.

Reply:
Oh okay. Still, I think it's better to manipulate the list from sys.path then, because for instance we're missing this dep now:

```
/home/raoul/.config/sublime-text-3/Packages/coverage/st3_linux_x64
```

Reply:
Should be resolved in https://github.com/sublimelsp/LSP-pyright/pull/26/commits/62c113d2d3dcfd828bd6120d5a1142c20fb2a3fe